### PR TITLE
Source Sonar Cloud: Add configurable API domain URL

### PR DESCRIPTION
**Pull request by shikanime**
Add configurable Sonar API URL

### CODE REVIEW

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml [hunk 1]
```diff
@@ -32,6 +42,13 @@ connectionSpecification:
       examples:
         - airbyte-ws-order
         - airbyte-ws-checkout
+    qualifiers:
+      title: Qualifiers
+      type: array
+      description: Comma-separated list of component qualifiers.
+      examples:
+        - TRK
+        - BRC
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:51]: Where this is used?

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml [hunk 2]
```diff
@@ -3,12 +3,22 @@ connectionSpecification:
...
+      - user_token
+      - qualifiers
+      - component_keys
   additionalProperties: true
   properties:
+    api_url:
+      type: string
+      title: Sonar API URL
+      description: Either SonarCloud or SonarQube API URL.
+      default: https://sonarcloud.io/api
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:21]: Can you explain why you need a configurable URL?

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml (hunk 2 above)
Comment (shikanime) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:21]: SonarQube is the open-source edition of the SonarCloud which company can selfhost.

**Comment by natikgadzhi** [comment]
Hey @shikanime, I'd love to get this in if I can help — would you mind rebasing this? We have a new shiny CI tool that I can kick-off and see if this is easily mergeable.

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml [hunk 3]
```diff
@@ -3,12 +3,22 @@ connectionSpecification:
...
-  required:
-    - user_token
-    - organization
-    - component_keys
+  oneOf:
+    - required:
+      - user_token
+      - organization
+      - component_keys
+    - required:
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:11]: Hmm, this looks weird here — why `oneOf` but of two identical things? 

Did you mean to have two exclusive options, one for cloud, and one for self-hosted set of config vars?

**Review by natikgadzhi — COMMENTED** [review]
Thank you for the PR! A couple of questions.

File: airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml (hunk 3 above)
Comment (shikanime) [reply airbyte-integrations/connectors/source-sonar-cloud/source_sonar_cloud/spec.yaml:11]: They aren't, the self-hosted version uses the 
qualifiers parameter, while the cloud-based version uses the organization parameter.

**Pull request closed without merging**